### PR TITLE
Move events to main thread and remove unneeded locks

### PR DESCRIPTION
**Pull request by andydotxyz**
This means for race-free apps except for data binding and goroutines, which are covered later, separately.

Progresses #4654

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: widget/widget.go [hunk 1]
```diff
@@ -19,9 +18,8 @@ type BaseWidget struct {
 	position async.Position
 	Hidden   bool
 
-	impl         atomic.Pointer[fyne.Widget]
-	propertyLock sync.RWMutex
-	themeCache   fyne.Theme
+	impl       atomic.Pointer[fyne.Widget]
```
Comment (Jacalz) [inline widget/widget.go:21]: Should atomic types and things from the `async` package also be removed or is that perhaps better to have as a separate PR?

**Review by Jacalz — COMMENTED** [review]
Nice work. Just one minor comment (not necessarily request for changes) inline. On more question: are there any parts where we should still support calling methods on different threads? Just asking so we don't get carried away and rip out too many locks at once :)

**Comment by Jacalz** [comment]
FYI: There seems to be compilation issues?

**Comment by andydotxyz** [comment]
> FYI: There seems to be compilation issues?

Yeah I just noteiced I didn't compile the tests for last refactor. Updated

File: widget/widget.go (hunk 1 above)
Comment (andydotxyz) [reply widget/widget.go:21]: I was wondering the same. I think later PR as it could impact semantics too

**Comment by andydotxyz** [comment]
> On more question: are there any parts where we should still support calling methods on different threads?

I think it's just data binding which requires more work for various reasons.

**Comment by andydotxyz** [comment]
Looks like there may be a deadlock on Linux only - I'll check it out tomorrow

**Review by Jacalz — APPROVED** [review]
Seems to work fine on Linux for me. Maybe deadlock is for tests only?

**Comment by andydotxyz** [comment]
> Seems to work fine on Linux for me. Maybe deadlock is for tests only?

Yes it was, or a codepath that was not executed on the testing. Fixed.

**Comment by andydotxyz** [comment]
One thing led to another and I just refactored the event queue out in this PR too.

Fixes a few flakey tests :)

**Pull request merged**